jobbergate-api/jobbergate_api/apps/job_script_templates/models.py

```diff
@@ -0,0 +1,134 @@
+from jobbergate_api.apps.models import Base, BaseFieldsMixin, Mapped, mapped_column
+
+
+class JobScriptTemplate(Base):
+    """
+    Job script template table definition.
+
+    Attributes:
+        id: The id of the job script template.
+        identifier: The identifier of the job script template.
+        name: The name of the job script template.
+        description: The description of the job script template.
+        owner_email: The email of the owner of the job script template.
+        template_vars: The template variables of the job script template.
+        created_at: The date and time when the job script template was created.
+        updated_at: The date and time when the job script template was updated.
+    """
+
+    __tablename__ = "job_script_templates"
+
+    id: Mapped[int] = mapped_column(Integer, primary_key=True)
+    identifier: Mapped[Optional[str]] = mapped_column(String, unique=True, index=True)
+    template_vars: Mapped[dict[str, Any]] = mapped_column(
+        "template_vars",
+        JSONB,
+        nullable=False,
+        default=text("'{}'::jsonb"),
+        server_default=text("'{}'::jsonb"),
+    )
+    name: Mapped[str] = mapped_column(String, nullable=False, index=True)
+    description: Mapped[Optional[str]] = mapped_column(String, default="")
+    owner_email: Mapped[str] = mapped_column(String, nullable=False, index=True)
+    created_at: Mapped[datetime] = mapped_column(DateTime, nullable=False, default=functions.now())
+    updated_at: Mapped[datetime] = mapped_column(
+        DateTime,
+        nullable=False,
+        default=functions.now(),
+        onupdate=functions.current_timestamp(),
+    )
+
+    template_files: Mapped[dict[str, "JobScriptTemplateFile"]] = relationship(
+        "JobScriptTemplateFile",
+        lazy="subquery",
+        collection_class=attribute_keyed_dict("filename"),
+        cascade="all, delete-orphan",
+    )
+    workflow_file: Mapped["WorkflowFile"] = relationship(
+        "WorkflowFile",
+        lazy="subquery",
+        cascade="all, delete-orphan",
+    )
+
+    searchable_fields = [
+        description,
+        id,
+        identifier,
+        name,
+        owner_email,
+    ]
+    sortable_fields = [
+        id,
+        name,
+        identifier,
+        owner_email,
+        created_at,
+        updated_at,
+    ]
+
+
+class JobScriptTemplateFile(Base, BaseFieldsMixin):
+    """Job script template files table definition."""
+
+    __tablename__ = "job_script_template_files"
+
+    id: Mapped[int] = mapped_column(
+        Integer,
+        ForeignKey(JobScriptTemplate.id, ondelete="CASCADE"),
+        primary_key=True,
+    )
+    filename: Mapped[str] = mapped_column(String, primary_key=True)
+    file_type: Mapped[FileType] = mapped_column(Enum(FileType), nullable=False)
+
+    @hybrid_property
+    def file_key(self) -> str:
+        return f"{self.__tablename__}/{self.id}/{self.filename}"
+
+
+class WorkflowFile(Base, BaseFieldsMixin):
```

Comment:
Smart Templates were renamed to workflow following a discussion on the data restructure proposal.

Template files can be rendered to job script files as long as rendering parameters are provided as we do from the UI. The Workflow file (jobbergate.py) is an alternative/optional way to do it from the CLI. So it made sense to centralize relevant columns on JobScriptTemplate in order to avoid repetition and have the WorkflowFIle attached to it.

I've created a diagram to illustrate:
https://gist.github.com/fschuch/51006fa5dc18ab24dee8533bf3fefe48

Of course, we can discuss it anytime.